### PR TITLE
Report FirebaseCore version

### PR DESCRIPTION
**Pull request by bstpierr**
Adds reporting of the FirebaseCore version as part of the useragent.

### CODE REVIEW

**Comment by ryanwilson** [comment]
@davidair I thought this was going to be for other libraries, not individual SDKs? Is this special because it's Core?

**Comment by paulb777** [comment]
Run ./scripts/style.sh to fix travis

**Comment by bstpierr** [comment]
Has the style script changed recently? It's telling me "Please upgrade to clang-format version 6.", but brew says I'm on the latest version.

**Comment by wilhuff** [comment]
You're on High Sierra, right? clang-format on high-sierra is now at 7.0.0

clang-format-2018-01-11.high_sierra.bottle.tar.gz ->
clang-format --version
clang-format version 7.0.0 (tags/google/stable/2018-01-11)

This is the same problem I had with swiftformat: High Sierra is now ahead of Sierra in homebrew because they're no longer updating Sierra.

**Comment by bstpierr** [comment]
Yep, I'm on High Sierra.

I locally skipped the version check for now.

File: Firebase/Core/FIROptions.m [hunk 1]
```diff
@@ -108,6 +109,21 @@ + (FIROptions *)defaultOptions {
 
 #pragma mark - Private class methods
 
++ (void)load {
+  // Report FirebaseCore version for useragent string
+  NSRange major = NSMakeRange(0, 1);
+  NSRange minor = NSMakeRange(1, 2);
+  NSRange patch = NSMakeRange(3, 2);
+  [FIRApp
+      registerLibrary:@"FirebaseCore"
```
Comment (davidair) [inline Firebase/Core/FIROptions.m]: Let's change this to "fire-ios" to make it compact and generic (other examples could be fire-unity and fire-cpp)

Comment (bstpierr) [reply Firebase/Core/FIROptions.m]: Done.

File: Firebase/Core/Private/FIRAppInternal.h [hunk 2]
```diff
@@ -134,16 +134,16 @@ typedef NSString *_Nullable (^FIRAppGetUIDImplementation)(void);
...
  * Registers a given third-party library with the given version number to be reported for
  * analyitcs.
  *
  * @param library Name of the library
  * @param version Version of the library
  */
-+ (void)registerLibrary:(nonnull NSString *)library withVersion:(nonnull NSString *)version
-    NS_SWIFT_NAME(registerLibrary(_:version:));
++ (void)registerLibrary:(nonnull NSString *)library
+            withVersion:(nonnull NSString *)version NS_SWIFT_NAME(registerLibrary(_:version
```
Comment (wilhuff) [inline Firebase/Core/Private/FIRAppInternal.h]: Unfortunately, clang-format can't handle `NS_SWIFT_NAME` annotations with more than one colon in them. Even more unfortunately, clang can't handle them when they're split across lines like this. As of now this is a broken build. The fix is to format this function declaration by hand and surround it with clang-format off/on, as done here:

https://github.com/firebase/firebase-ios-sdk/blob/90dedd3d024a2859d0cf514efbfc97e471891c18/Firestore/Source/Public/FIRQuery.h#L111

Comment (bstpierr) [reply Firebase/Core/Private/FIRAppInternal.h]: Thanks for the tip! I was looking at this earlier and wonder what happened. I'll format it manually with the format comments.

**Pull request merged**